packages/react/src/template.tsx

```diff
@@ -117,36 +130,49 @@ export function BaseTemplate({
   dataState: DataState
   onTemplateEvent?: (event: TemplateEvent) => any
 }) {
-  function renderComponent(stateNode: ComponentDataState, path: Path) {
-    const componentDefinition = components[stateNode.component]
-    if (!componentDefinition) {
-      throw new RenderError(`Unknown component: ${stateNode.component}`)
-    }
-
-    // tbd: validate `components` prop once with `zod` instead of later render stage
-    const Component = componentDefinition.component
-    if (!Component) {
-      throw new RenderError(`Invalid component: ${stateNode.component}`)
-    }
+  const RenderComponent = useCallback(
```

Comment:
This was causing TextInput to loose focus, because this function was re-created each render, resulting in a whole new component being put on screen (unmount input, mount a new one). Likely the reason why some BottomSheet issues were happening!

Reply:
oh!! I knew it would be something simple! thank you for finding this, it has caused a lot of pain